platform/posix/transport/src/openssl_posix.c

```diff
@@ -166,10 +166,7 @@ static OpensslStatus_t convertToOpensslStatus( SocketStatus_t socketStatus );
         }
 
         /* Free cwd because getcwd calls malloc. */
-        if( cwd != NULL )
-        {
-            free( cwd );
-        }
+        free( cwd );
```

Comment:
Calling a `free` on a NULL pointer is fine, so removing the check leads to cleaner code.